### PR TITLE
wip: use rdfjs vc library

### PR DESCRIPTION
**Pull request by jeswr**
This PR fixes #<issue ID>.

- [ ] I've added a unit test to test for potential regressions of this bug.
- [ ] The changelog has been updated, if applicable.
- [ ] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

# New feature description

# Checklist

- [ ] All acceptance criteria are met.
- [ ] Relevant documentation, if any, has been written/updated.
- [ ] The changelog has been updated, if applicable.
- [ ] New functions/types have been exported in `index.ts`, if applicable.
- [ ] New modules (i.e. new `.ts` files) are listed in the `exports` field in `package.json`, if applicable.
- [ ] New modules (i.e. new `.ts` files) are listed in the `typedocOptions.entryPoints` field in `tsconfig.json`, if applicable.
- [ ] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

This PR bumps the version to <version number>.

# Release steps:

1. Inspecting the `CHANGELOG.md` to determine if the release was major, minor or patch. Make sure you align with the rest of the team first.
2. Use `npm version --no-git-tag-version <major|minor|patch>` to update `package.json`.
3. Update the `CHANGELOG.md` has been updated to show version and release date.
4. Add any `@since X.Y.Z` annotations if there have been any added or deprecated APIs.
5. Commit the changes as `release: prepare vX.Y.Z` on a branch named `release/X.Y.Z`
6. Create a PR, once CI passes, merge as a squash merge.
7. Create and publish the release via GitHub, this will create the tag and trigger the npm publish.

### CODE REVIEW

File: src/gConsent/manage/approveAccessRequest.test.ts [hunk 1]
```diff
@@ -65,12 +65,26 @@ jest.mock("@inrupt/solid-client", () => {
...
   return {
-    ...vcModule,
+    verifiableCredentialToDataset,
+    getCredentialSubject,
+    getIssuanceDate,
+    getIssuer,
+    getExpirationDate,
+    getId,
     issueVerifiableCredential: jest.fn(),
+    getVerifiableCredential,
```
Comment (NSeydoux) [inline src/gConsent/manage/approveAccessRequest.test.ts]: Why this change?

File: src/gConsent/manage/approveAccessRequest.ts [hunk 2]
```diff
@@ -305,29 +453,17 @@ export async function approveAccessRequest(
       );
     }
 
-    return accessGrant;
+    return accessGrant as AccessGrant;
```
Comment (NSeydoux) [inline src/gConsent/manage/approveAccessRequest.ts:456]: Why is the type assertion required despite the type guards above?

File: src/resource/saveSolidDatasetAt.ts [hunk 3]
```diff
@@ -47,7 +50,7 @@ import type { FetchOptions } from "../type/FetchOptions";
 export async function saveSolidDatasetAt<Dataset extends SolidDataset>(
   datasetUrl: UrlString,
   solidDataset: Dataset,
-  accessGrant: VerifiableCredential,
+  accessGrant: VerifiableCredential | DatasetWithId,
```
Comment (NSeydoux) [inline src/resource/saveSolidDatasetAt.ts:53]: In other instances in this PR, `VerifiableCredential` has been replaced by `DatasetWithId`. Why the union here?

File: src/common/getters.ts [hunk 4]
```diff
@@ -18,190 +18,371 @@
...
  * @param vc The Access Grant/Request
- * @returns The requestor WebID
+ * @returns The resources IRIs
  */
-export function getRequestor(
-  vc: AccessGrantGConsent | AccessRequestGConsent,
-): string {
-  if (isGConsentAccessGrant(vc)) {
-    return vc.credentialSubject.providedConsent.isProvidedTo;
+export function getResources(vc: DatasetWithId): string[] {
```
[the remark this answers is not in the document]
Comment (jeswr) [reply src/common/getters.ts:162]: This is correct :)

File: src/gConsent/guard/isAccessGrantContext.ts [hunk 5]
```diff
@@ -22,6 +22,11 @@
 import { CONTEXT_VC_W3C } from "../constants";
 import type { AccessGrantContext } from "../type/AccessGrantContext";
 
+/**
+ *
+ * @param x @deprecated
+ * @returns
+ */
```
Comment (jeswr) [inline src/gConsent/guard/isAccessGrantContext.ts:29]:
```suggestion
/**
 * @deprecated
 */
```

File: src/gConsent/guard/isGConsentAttributes.ts [hunk 6]
```diff
@@ -50,3 +60,59 @@ export function isGConsentAttributes(
...
+    consent,
+    gc.forPersonalData,
+    null,
+    defaultGraph(),
+  );
+
+  // FIXME: Should we really have this condition?
+  if (forPersonalData.size === 0) {
+    throw new Error("No Personal Data specified for Access Grant");
+  }
```
Comment (jeswr) [inline src/gConsent/guard/isGConsentAttributes.ts:109]: FYI @NSeydoux this condition means that this function will error if the access grant has no resources.

File: src/resource/saveSolidDatasetAt.ts (hunk 3 above)
Comment (jeswr) [reply src/resource/saveSolidDatasetAt.ts:53]: The `VerifiableCredential` is not necessary from a type safety perspective as it is a superset of `DatasetWithId` (as are `AccessGrant` and `AccessRequest`). These are only left in the type union to make it more readable for humans looking at the types.

File: src/gConsent/manage/approveAccessRequest.ts (hunk 2 above)
Comment (jeswr) [reply src/gConsent/manage/approveAccessRequest.ts]: Good catch - this is indeed no longer necessary.

File: src/gConsent/guard/isGConsentAttributes.ts (hunk 6 above)
Comment (jeswr) [reply src/gConsent/guard/isGConsentAttributes.ts:109]: We used to check if there was an array. I think it was assumed that it was non-empty if one was present; but was didn't explicitly check this.

**Comment by chelseapinka** [comment]
Is "wip" still accurate this does not need review yet?

**Comment by jeswr** [comment]
This is wip because the vc and access grants library are still in beta.

I will mark this as ready for review once they have a full release.

**Comment by NSeydoux** [comment]
Superceded by #875
